src/components/common/EdgeCarousel.tsx

```diff
@@ -24,14 +24,25 @@ export function EdgeCarousel<T>(props: Props<T>): JSX.Element {
   const theme = useTheme()
   const styles = getStyles(theme)
 
-  const carouselRef = React.useRef(null)
+  const carouselRef = React.useRef<Carousel<any>>(null)
 
   const [activeIndex, setActiveIndex] = useState(0)
 
   const renderItem = useHandler<ListRenderItem<T>>(info => (
     <View style={[styles.childContainer, { width: width * 0.9, height }]}>{props.renderItem(info)}</View>
   ))
 
+  /**
+   * Carousel's FlatList bug workaround
+   */
+  React.useEffect(() => {
+    if (carouselRef.current != null) {
+      setTimeout(() => {
```

Comment:
Seems like you want something like `runAfterInteractions` or `requestAnimationFrame` which is less hacky and not prone to race conditions.